openmc/deplete/abc.py

```diff
@@ -160,3 +163,165 @@ def get_results_info(self):
+        consistent with :class:`openmc.deplete.Operator`
+    """
+
+    def __init__(self):
+        self._nuclides = None
+        self._rate_tally = None
+        self._results_cache = None
+
+    @abstractmethod
+    def generate_tallies(self, materials, scores):
+        """Use the capi to build tallies needed for reaction rates"""
+
+    @property
+    def nuclides(self):
+        """List of nuclides with requested reaction rates"""
+        return self._nuclides
+
+    @nuclides.setter
+    def nuclides(self, nuclides):
+        check_type("nuclides", nuclides, list, str)
+        self._nuclides = nuclides
+        self._rate_tally.nuclides = nuclides
+
+    def _reset_results_cache(self, nnucs, nreact):
+        """Cache for results for a given material"""
+        if (self._results_cache is None
+                or self._results_cache.shape != (nnucs, nreact)):
+            self._results_cache = zeros((nnucs, nreact))
+        else:
+            self._results_cache.fill(0.0)
+        return self._results_cache
+
+    @abstractmethod
+    def get_material_rates(self, mat_id, nuc_index, react_index):
+        """Return 2D array of [nuclide, reaction] reaction rates
+
+        Parameters
+        ----------
+        nuc_index : list of str
+            Ordering of desired nuclides
+        react_index : list of str
+            Ordering of reactions
+        """
+
+    def divide_by_adens(self, number):
+        """Normalize reaction rates by number of nuclides
+
+        Acts on the current material examined by
+        :meth:`get_material_rates`
+
+        Parameters
+        ----------
+        energy : float
+            Energy produced in this region [W]
+        number : iterable of float
+            Number density [atoms/b/cm] of each nuclide tracked in the calculation.
+            Ordered identically to :attr:`nuclides`
+
+        Returns
+        -------
+        results : `numpy.ndarray`
+            2D array ``[n_nuclides, n_rxns]`` of reaction rates normalized by
+            the number of nuclides
+        """
+
+        mask = nonzero(number)
+        results = self._results_cache
+        for col in range(results.shape[1]):
+            results[mask, col] /= number[mask]
+        return results
+
+
+class EnergyHelper(ABC):
+    """Abstract class for obtaining energy produced
+
+    The ultimate goal of this helper is to provide instances of
+    :class:`openmc.deplete.Operator` with the total energy produced
+    in a transport simulation. This information, provided with the
+    power requested by the user and reaction rates from a
+    :class:`ReactionRateHelper` will scale reaction rates to the
+    correct values.
+
+    Attributes
+    ----------
+    nuclides : list of str
+        All nuclides with desired reaction rates. Ordered to be
+        consistent with :class:`openmc.deplete.Operator`
+    energy : float
+        Total energy [eV/s] produced in a transport simulation.
```

Comment:
There also needs to be a "per source neutron" in the units. Also, perhaps we should just return a value with units of J rather than eV?

Reply:
Easy enough. The conversion has been moved to the `energy` property, and the conversion from power in W to eV/s has been removed